src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -806,24 +779,20 @@ private void ParseHelper(string[] args)
                     ++i;
                     if (i >= args.Length)
                     {
-                        WriteCommandLineError(
+                        SetCommandLineError(
                             CommandLineParameterParserStrings.MissingCustomPipeNameArgument);
                         break;
                     }
 
-                    if (!Platform.IsWindows)
+                    if (!Platform.IsWindows && args[i].Length > MaxNameLength())
```

Comment:
Don't call `MaxNameLength()` twice. Use a variable within `if (!Platform.IsWindows)`, just like before you change.

Reply:
Done.